builder/tests/integration_tests.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    fs,
+    net::{Ipv4Addr, SocketAddr},
+    path::Path,
+};
+
+use portpicker::pick_unused_port;
+use pretty_assertions::assert_eq;
+use shuttle_builder::Service;
+use shuttle_common::claims::Scope;
+use shuttle_common_tests::JwtScopesLayer;
+use shuttle_proto::builder::{
+    build_response::Secret, builder_client::BuilderClient, builder_server::BuilderServer,
+    BuildRequest,
+};
+use tokio::select;
+use tonic::{transport::Server, Request};
+use ulid::Ulid;
+
+#[tokio::test]
+async fn build_crate() {
+    let port = pick_unused_port().unwrap();
+    let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
+
+    let server_future = async {
+        Server::builder()
+            .layer(JwtScopesLayer::new(vec![Scope::DeploymentPush]))
+            .add_service(BuilderServer::new(Service::new()))
+            .serve(addr)
+            .await
+            .unwrap()
+    };
+
+    let test_future = async {
+        let mut client = BuilderClient::connect(format!("http://localhost:{port}"))
+            .await
+            .unwrap();
+        let resources = Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("tests")
+            .join("resources");
+
+        // Build a normal hello world archive
+        let deployment_id = Ulid::new().to_string();
+        let archive = fs::read(resources.join("hello-world-0.1.0.crate")).unwrap();
```

Comment:
Oh, does the new builder take .crate files? Does that mean we need to follow the cargo package restrictions again?

Reply:
Oh, no. A `.crate` is just a `tar.gz`. I'll update the test :sweat_smile: